### PR TITLE
Added mailer for Broker approval

### PR DESCRIPTION
**Pull request by jeleencubillas**
## The Story
- The User Story requires Admins to approve Brokers before they can access their account
- We added a boolean field to Users that defaults to "true" when the user signs up as Buyer, and "false" when the User signs up as Broker
- We created an ActionMailer that sends a notification to Admin when the user signs up as Broker
- We also created a Users index so Admins can view all users. All views will be styled later on.

## Screenshots
image
User Confirmation Email

image
Broker Approval Email to Admin

image
User Model Tests

## Commit Messages 
- Added mailer for Broker approval
- Added empty last line to new files

## References
https://github.com/heartcombo/devise/wiki/How-to-Setup-Multiple-Devise-User-Models
https://github.com/heartcombo/devise/wiki/I18n
https://guides.rubyonrails.org/action_mailer_basics.html
https://chelseatroy.com/2019/04/08/modifying-authentication-behavior-in-devise/

## What's Missing
- [ ] Add Deployed Link to Admin Approval Email
- [ ] Views Styling

### CODE REVIEW

File: app/controllers/concerns/accessible.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        flash.clear
+        # if you have rails_admin. You can redirect anywhere really
+        redirect_to(rails_admin.dashboard_path) and return
+      elsif current_user
+        flash.clear
+        # The authenticated root path can be defined in your routes.rb in: devise_scope :user do...
+        redirect_to(root_path) and return
+      end
+    end
+  end
```
Comment (analyn34) [inline app/controllers/concerns/accessible.rb:19]: add newline at end of file

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  
+  def index
+    if params[:approved] == "false"
+      @users = User.where(approved: false)
+    else
+      @users = User.all
+    end
+  end
+
+end
```
Comment (analyn34) [inline app/controllers/users_controller.rb:11]: add newline at end of file

File: app/controllers/concerns/accessible.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
```
Comment (analyn34) [inline app/controllers/concerns/accessible.rb:19]: fix code alignment

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<table>
+    <% @users.each do |user| %>
+      <tr>
+        <td><%= user.email %>
+        <td><%= user.role %>
+        <td><%= user.approved.to_s %>
+      </tr>
+    <% end %>
+</table>
```
Comment (analyn34) [inline app/views/users/index.html.erb:9]: add newline at end of file

File: app/controllers/concerns/accessible.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    included do
+      before_action :check_user
+    end
+  
+    protected
+    def check_user
+      if current_admin
+        flash.clear
+        # if you have rails_admin. You can redirect anywhere really
+        redirect_to(rails_admin.dashboard_path) and return
```
Comment (analyn34) [inline app/controllers/concerns/accessible.rb:12]: Did you use `rails_admin` ?

Comment (jeleencubillas) [reply app/controllers/concerns/accessible.rb]: No Maam, will resolve in the next commit

**Review by analyn34 — APPROVED** [review]

**Pull request merged**